### PR TITLE
Include the error message as well as error code in the error message

### PR DESCRIPTION
**Pull request by fredzqm**
Tested here. Having the error message makes it so much easy to tell what happened.

We can polish up additional error handling later.

To prod:
Screenshot 2024-09-27 at 11 19 18 AM

To emulator:
Screenshot 2024-09-27 at 11 23 33 AM

### CODE REVIEW

File: firebase-vscode/src/data-connect/service.ts [hunk 1]
```diff
@@ -243,23 +234,23 @@ export class DataConnectService {
...
+      const endpoint = this.emulatorsController.getLocalEndpoint();
+      if (!endpoint) {
+        throw new DataConnectError(
+          `Emulator isn't running. Please start your emulator!`,
+        );
+      }
+      const client = new Client({
+        urlPrefix: endpoint,
+        apiVersion: "v1beta",
+        auth: true,
```
Comment (yuchenshi) [inline firebase-vscode/src/data-connect/service.ts:250]: Please revert this to use `"x-mantle-admin": "all"` instead. `auth: true` may send production auth tokens to the emulator and may run into issues when developing offline.

If the `Client` API doesn't do what we want, then using `fetch` is also okay for now. We can always clean code up later but let's not mess with the behavior.

File: firebase-vscode/src/data-connect/service.ts [hunk 2]
```diff
@@ -78,47 +81,35 @@ export class DataConnectService {
...
-    assertExecutionResult(json);
-
-    return json;
-  }
-
-  private async handleInvalidResponse(response: Response): Promise<never> {
-    const cause = await this.decodeResponse(response);
-
-    throw new DataConnectError(
-      `Request failed with status ${response.status}`,
```
Comment (fredzqm) [inline firebase-vscode/src/data-connect/service.ts:111]: It swallowed the error message here.

File: firebase-vscode/src/data-connect/service.ts [hunk 3]
```diff
@@ -243,23 +234,23 @@ export class DataConnectService {
...
-      return this.handleResponse(resp);
+      const endpoint = this.emulatorsController.getLocalEndpoint();
+      if (!endpoint) {
+        throw new DataConnectError(
+          `Emulator isn't running. Please start your emulator!`,
+        );
+      }
+      const client = new Client({
+        urlPrefix: endpoint,
+        apiVersion: "v1beta",
```
Comment (joehan) [inline firebase-vscode/src/data-connect/service.ts:249]: Probably should use DATACONNECT_API_VERSION here

File: firebase-vscode/src/data-connect/service.ts [hunk 4]
```diff
@@ -78,47 +81,35 @@ export class DataConnectService {
...
       ExecuteGraphqlResponse | ExecuteGraphqlResponseError
     >,
   ): Promise<ExecutionResult> {
-    if (!(clientResponse.status >= 200 && clientResponse.status < 300)) {
+    if (!(response.status >= 200 && response.status < 300)) {
       const errorResponse =
-        clientResponse as ClientResponse<ExecuteGraphqlResponseError>;
+        response as ClientResponse<ExecuteGraphqlResponseError>;
       throw new DataConnectError(
-        `Request failed with status ${clientResponse.status}`,
```
Comment (fredzqm) [inline firebase-vscode/src/data-connect/service.ts:89]: It swallowed the error message here.

**Review by yuchenshi — APPROVED** [review]

**Review by joehan — APPROVED** [review]

File: firebase-vscode/src/data-connect/service.ts (hunk 1 above)
Comment (fredzqm) [reply firebase-vscode/src/data-connect/service.ts:250]: What about remove `auth: true` here?

`x-mantle-admin` doesn't matter.

File: firebase-vscode/src/data-connect/service.ts (hunk 3 above)
Comment (fredzqm) [reply firebase-vscode/src/data-connect/service.ts]: Done

File: firebase-vscode/src/data-connect/service.ts (hunk 1 above)
Comment (fredzqm) [reply firebase-vscode/src/data-connect/service.ts]: I tested executing against emulator. Everything works fine, both error and success.

Comment (fredzqm) [reply firebase-vscode/src/data-connect/service.ts]: Here are some emulator logs: https://paste.googleplex.com/6332029630414848

Good catch on not sending auth token. Emulator has V(3) debug logs for the auth token, won't log by default even if someone sends a real token to it.

**Pull request merged**